src/main/scala/org/scalacheck/rng/Seed.scala

```diff
@@ -99,6 +99,9 @@ sealed abstract class Seed extends Serializable {
 object Seed {
 
   private case class apply(a: Long, b: Long, c: Long, d: Long) extends Seed {
+    if (a == 0 && b == 0 && c == 0 && d == 0) {
+      throw new IllegalArgumentException("illegal Seed(0, 0, 0, 0)")
```

Comment:
Why not check this in `fromLongs`?

Reply:
I actually discovered this via with a [magnolia](https://github.com/propensive/magnolia) derivation of `Arbitrary[T]`. Turns out magnolia can derive the `sealed abstract class Seed` by calling its `private case class apply(...)`.

Derivation code:
https://github.com/spotify/magnolify/blob/master/scalacheck/src/main/scala/magnolify/scalacheck/semiauto/ArbitraryDerivation.scala#L25

This calls `apply()`
```
implicitly[Arbitrary[Seed]].arbitrary.sample
```

Checking in `apply` would prevent these macro libraries from accidentally deriving a bad seed.

Reply:
Is it calling `private case class apply` or its own subclass of `Seed`?

It shouldn't be able to do either, as `Seed` is `sealed` and `apply` is `private`, so I'd argue that particular behaviour is a problem that needs resolving in Magnolia.  (But I think this PR has merit regardless.)

Reply:
Yeah looks like magnolia is able to call the private case class thru macros.

I admit that doing the check here might add a bit of overhead, since `Seed#apply(s: Long)` does this `while (i < 20) { seed = seed.next; i += 1 }` which calls `apply()`.
I can probably override the implicit in my library, but such a check is still nice to have, maybe in `fromLongs` as you suggested?

![dbg](https://user-images.githubusercontent.com/468514/89804303-fd56dd00-db01-11ea-8d6b-bca38de92da9.png)

Reply:
So, the intention was for this constructor to be private -- the issue of an all-zero seed is known and the code was being defensive to avoid it.

I'm fine adding this check to `fromLongs` but I'd rather not require every (internal) use of `apply` to do the check.

Reply:
Cool, my thought as well. Will move it to `fromLongs` and fix it on my end to avoid `apply(0, 0, 0, 0)`.

Reply:
@non moved to `fromLongs` PTAL thanks :)